Bottom: d9c2e699f1e1ba4e8594d75b2371252c9669f3ca
Top:    bba2ec32fc04c8b02e84fb3d865c315938629cb2
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-06-28 15:59:01 +0800

virtio-net: do not reset during xdp set

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 54f33ce..772bd9d 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -1942,6 +1942,7 @@ static int virtnet_xdp_set(struct net_device *dev, struct bpf_prog *prog,
 			return PTR_ERR(prog);
 	}
 
+#if 0
 	/* Changing the headroom in buffers is a disruptive operation because
 	 * existing buffers must be flushed and reallocated. This will happen
 	 * when a xdp program is initially added or xdp is disabled by removing
@@ -1954,6 +1955,7 @@ static int virtnet_xdp_set(struct net_device *dev, struct bpf_prog *prog,
 			goto virtio_reset_err;
 		}
 	}
+#endif
 
 	netif_set_real_num_rx_queues(dev, curr_qp + xdp_qp);
